batch/sql/estimated-current.sql

```diff
@@ -486,23 +537,63 @@ BEGIN
     WHERE batch_id = NEW.batch_id AND job_id = NEW.job_id AND attempt_id = NEW.attempt_id
     ON DUPLICATE KEY UPDATE `usage` = `usage` + msec_diff_rollup * quantity;
 
+    INSERT INTO aggregated_billing_project_user_resources_v3 (billing_project, user, resource_id, token, `usage`)
+    SELECT batches.billing_project, batches.`user`,
+      attempt_resources.resource_id,
+      rand_token,
+      msec_diff_rollup * quantity
+    FROM attempt_resources
+    JOIN batches ON batches.id = attempt_resources.batch_id
+    INNER JOIN aggregated_billing_project_user_resources_v2 ON
+      aggregated_billing_project_user_resources_v2.billing_project = batches.billing_project AND
+      aggregated_billing_project_user_resources_v2.user = batches.user AND
+      aggregated_billing_project_user_resources_v2.resource_id = attempt_resources.resource_id AND
+      aggregated_billing_project_user_resources_v2.token = rand_token
+    WHERE attempt_resources.batch_id = NEW.batch_id AND attempt_resources.job_id = NEW.job_id AND attempt_id = NEW.attempt_id AND migrated = 1
+    ON DUPLICATE KEY UPDATE `usage` = aggregated_billing_project_user_resources_v3.`usage` + msec_diff_rollup * quantity;
+
     INSERT INTO aggregated_batch_resources_v2 (batch_id, resource_id, token, `usage`)
-    SELECT attempt_resources.batch_id,
+    SELECT batch_id,
       resource_id,
       rand_token,
       msec_diff_rollup * quantity
     FROM attempt_resources
     WHERE batch_id = NEW.batch_id AND job_id = NEW.job_id AND attempt_id = NEW.attempt_id
     ON DUPLICATE KEY UPDATE `usage` = `usage` + msec_diff_rollup * quantity;
 
+    INSERT INTO aggregated_batch_resources_v3 (batch_id, resource_id, token, `usage`)
+    SELECT attempt_resources.batch_id,
+      attempt_resources.resource_id,
```

Comment:
Note that in these triggers, we're using the resource_id instead of the deduped_resource_id. At this point, any new resource ids always equal the deduped resource id because we already performed the mitigation to not add dups to the database and there's a complete resource dedup map. When we eventually rename the columns from resource_id -> deduped_resource_id and deduped_resource_id to resource_id when we want to change the primary key to use the "deduped resource id" for the attempt resources table, the rename operation only changes the column references for indices and foreign key constraints. It does not change the name or reference to of the column in the triggers or stored procedures. So it's best to keep this to be the `resource_id` here so as to be able to drop the old "resource_id" column later on.

Reply:
So to parrot this back, use `resource_id` when the two are equivalent because it will be the permanent name in the long-run?

Reply:
yes. exactly. It does make me uneasy to do this, but I'm worried about what happens when we try and cleanup the attempt resources table. This seemed better than having the triggers temporarily referencing a column that does not exist. But maybe I'm overthinking it.

Reply:
Ok. I made things as clear as possible referencing `deduped_resource_id` where possible. For these specific queries, however, we cannot assume that the deduped resource id has already been populated as that only applies to new attempts that were generated after #12028 has fully redeployed. Therefore, we use the `resource_id` as that will never be NOT NULL and should be already deduped post-mitigation.